playbook.yaml

```diff
@@ -26,14 +26,15 @@
   roles:
     - { role: validate-input, tags: ['validate-input'] }
     - { role: validate-input-apigee-control-plane, tags: ['validate-input', 'validate-input-apigee'] }
-    - { role: validate-input-k8s-cluster, tags: ['validate-input', 'validate-input-k8s'] }
+    - { role: validate-input-k8s-cluster, vars: { custom_cassandra_storage_class: '{{ overrides.cassandra.storage.storageClass }}' }, tags: ['validate-input', 'validate-input-k8s'] }
     - { role: cert-manager, tags: ['cert-manager'] }
     - { role: certificates, vars: { generate_certificates: true }, tags: ['certs'] }
     - { role: prepare-helm, tags: ['prepare-helm'] }
     - { role: prepare-service-accounts, tags: ['prepare-service-accounts'] }
     - { role: enable-apigee-synchronizer, tags: ['enable-synchronizer'] }
     - { role: bootstrap-apigee-crds, tags: ['bootstrap-apigee-crds'] }
     - { role: apigee-hybrid-overrides, tags: ['generate-overrides'] }
+    - { role: apigee-custom-sc, vars: { storageClassName: '{{ overrides.cassandra.storage.storageClass }}', storageClassType: '{{ overrides.cassandra.storage.storageClassProperties.provisionerType }}', scParameters: '{{ overrides.cassandra.storage.storageClassProperties.parameters }}' }, tags: ['create-custom-sc'] }
```

Comment:
@RiflerRick : similar to https://github.com/apigee/ansible-apigee-hybrid-accelerator/pull/31/files#r1441612629

Reply:
sure

Reply:
actually, moving the global variable references inside the role will tightly couple it. if we decide to change the structure of the variable in the vars file, we will have to start changing the roles too. The role IMO should have a clean interface. Let me know what you think
@anaik91